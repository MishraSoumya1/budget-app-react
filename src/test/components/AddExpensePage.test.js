import React from "react";
import { shallow } from "enzyme";
import { AddExpensePage } from "../../components/AddExpensePage";
import { expenses } from "../fixtures/expenseData";

let onSubmit, history, wrapper;

beforeEach(() => {
  onSubmit = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<AddExpensePage onSubmit={onSubmit} history={history} />);
});

describe("Validating add expense page", () => {
  test("validate add expense functionality", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should handle onSubmit", () => {
    wrapper.find("ExpenseForm").prop("onSubmit")({
      description: "Gum",
      note: "",
      amount: 195,
      createdAt: 0,
    });
    expect(history.push).toHaveBeenLastCalledWith("/");
    expect(onSubmit).toHaveBeenLastCalledWith({
      description: "Gum",
      note: "",
      amount: 195,
      createdAt: 0,
    });
  });
});
