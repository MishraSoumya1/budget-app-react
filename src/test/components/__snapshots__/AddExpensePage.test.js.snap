// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validating add expense page validate add expense functionality 1`] = `
<div>
  <h1>
    Add Expense
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
